### ISSUE TITLE
unarchive module is not able to unpack zip archive containing files with spaces in names (2.1.0)

### ISSUE THREAD

**Issue by risobalaz** [ansible/ansible-modules-core#3792]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
forks          = 10
gather_subset = !all
timeout = 30

[privilege_escalation]
become_allow_same_user = True

[ssh_connection]
ssh_args = -o ControlMaster=auto -o ControlPersist=60s
pipelining = True
scp_if_ssh = True
```
##### OS / ENVIRONMENT

Oracle Linux Server release 7.1 (both master and slave)
##### SUMMARY

unarchive module ends with error if zip archive contains files with spaces in names
##### STEPS TO REPRODUCE

```
# unzip archive.zip
Archive:  archive.zip
   creating: archive/
  inflating: archive/file 1.txt
  inflating: archive/file 2.txt
  inflating: archive/file 3.txt

# ansible -i inventory.ini security_nodes -m unarchive -a "src=archive.zip dest=/tmp"
```
##### EXPECTED RESULTS

Archive is properly unpacked.
##### ACTUAL RESULTS

```
# ansible -i inventory.ini security_nodes -m unarchive -a "src=archive.zip dest=/tmp" -vvvv
Using /etc/ansible/ansible.cfg as config file
Loaded callback minimal of type stdout, v2.0
<ci-security> ESTABLISH SSH CONNECTION FOR USER: None
<ci-security> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=30 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r ci-security '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679 `" && echo ansible-tmp-1464613962.55-9622357930679="` echo $HOME/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679 `" ) && sleep 0'"'"''
<ci-security> ESTABLISH SSH CONNECTION FOR USER: None
<ci-security> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=30 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r ci-security '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<ci-security> PUT /tmp/ansible/archive.zip TO /root/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679/source
<ci-security> SSH: EXEC scp -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=30 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r /tmp/ansible/archive.zip '[ci-security]:/root/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679/source'
<ci-security> ESTABLISH SSH CONNECTION FOR USER: None
<ci-security> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=30 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r ci-security '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python && sleep 0'"'"''
<ci-security> ESTABLISH SSH CONNECTION FOR USER: None
<ci-security> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=30 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r ci-security '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679/ > /dev/null 2>&1 && sleep 0'"'"''
ci-security | FAILED! => {
    "changed": false,
    "dest": "/tmp",
    "failed": true,
    "gid": 0,
    "group": "root",
    "handler": "ZipArchive",
    "invocation": {
        "module_args": {
            "backup": null,
            "content": null,
            "copy": true,
            "creates": null,
            "delimiter": null,
            "dest": "/tmp",
            "directory_mode": null,
            "exclude": [],
            "extra_opts": [],
            "follow": false,
            "force": null,
            "group": null,
            "keep_newer": false,
            "list_files": false,
            "mode": null,
            "original_basename": "archive.zip",
            "owner": null,
            "regexp": null,
            "remote_src": null,
            "selevel": null,
            "serole": null,
            "setype": null,
            "seuser": null,
            "src": "/root/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679/source",
            "validate_certs": true
        }
    },
    "mode": "01777",
    "msg": "Unexpected error when accessing exploded file: [Errno 2] No such file or directory: '/tmp/archive/file 1.txt'",
    "owner": "root",
    "size": 4096,
    "src": "/root/.ansible/tmp/ansible-tmp-1464613962.55-9622357930679/source",
    "state": "directory",
    "uid": 0
}
```

**Comment by heyitsmdr** [comment]
We just noticed this a few days ago as well. I'm going to submit a PR to fix this. The bug is located here:

https://github.com/ansible/ansible-modules-core/blob/devel/files/unarchive.py#L316

The issue is that it is iterating over each line returned from `unzip -ZT <source>` and splitting the line at each space. This will return an array with 8 members unless the path contains a space, in which case it will contain 9+. Here is an example of the `unzip -ZT` output:

```
$ unzip -ZT test.zip
Archive:  test.zip   23242 bytes   5 files
-rw-r--r--  2.1 unx    10939 bX defN 20160524.194352 no_spaces.png
drwxrwxr-x  2.1 unx        0 bx stor 20160526.095212 __MACOSX/
-rw-r--r--  2.1 unx      423 bX defN 20160524.194352 __MACOSX/._no_spaces.png
-rw-r--r--  2.1 unx    10939 bX defN 20160524.194352 with spaces.png
-rw-r--r--  2.1 unx      423 bX defN 20160524.194352 __MACOSX/._with spaces.png
5 files, 22724 bytes uncompressed, 22462 bytes compressed:  1.2%
```

**Comment by bcoca** [comment]
closing via merged #3819
